CollapseLauncher/Classes/GameManagement/GameSettings/Honkai/RegistryClass/GraphicsGrade.cs

```diff
@@ -0,0 +1,85 @@
+
+        #region Properties
+        /// <summary>
+        /// This defines the "<c>Video -> Graphics</c>" Quality preset Combobox <br/><br/>
+        /// Value is int with possible value of 1 - 6
+        /// 
+        /// </summary>
+        public SelectGraphicsGrade GraphicsGradeInt { get; set; } = SelectGraphicsGrade.Custom;
+        #endregion
+        
+        #region Methods
+        #nullable enable
+        public static GraphicsGrade Load()
+        {
+            try
+            {
+                if (RegistryRoot == null) throw new NullReferenceException($"Cannot load {_ValueName} RegistryKey is unexpectedly not initialized!");
+
+                object? value = RegistryRoot.GetValue(_ValueName, null);
+                if (value != null)
+                {
+                    SelectGraphicsGrade graphicsGrade = (SelectGraphicsGrade)value;
+                    #if DEBUG
+                    LogWriteLine($"Loaded HI3 Settings: {_ValueName} : {value}", LogType.Debug, true);
+                    #endif
+
+                    return new GraphicsGrade { GraphicsGradeInt = graphicsGrade };
+                }
+            }
+            catch ( Exception ex )
+            {
+                LogWriteLine($"Failed while reading {_ValueName}" +
+                                $"\r\n  Please open the game and change any Graphics Settings, then close normally. After that you can use this feature." +
+                                $"\r\n  If the issue persist, please report it on GitHub" +
+                                $"\r\n{ex}", LogType.Error, true);
+                ErrorSender.SendException(new Exception(
+                            $"Failed when reading game settings {_ValueName}\r\n" +
+                            $"Please open the game and change any graphics settings, then safely close the game. If the problem persist, report the issue on our GitHub\r\n" +
+                            $"{ex}", ex));
+            }
+
+            return new GraphicsGrade();
+        }
+
+        public void Save()
+        {
+            try
+            {
+                if (RegistryRoot == null) throw new NullReferenceException($"Cannot save {_ValueName} since RegistryKey is unexpectedly not initialized!");
+                LogWriteLine("(HI3 GSP) Forcing GraphicsGrade to Custom!", LogType.Warning, true);
+                GraphicsGradeInt = SelectGraphicsGrade.Custom;
+                RegistryRoot.SetValue(_ValueName, GraphicsGradeInt, Microsoft.Win32.RegistryValueKind.DWord);
+                #if DEBUG
+                LogWriteLine($"Saved HI3 Settings: {_ValueName} : {GraphicsGradeInt}", LogType.Debug, true);
+                #endif
+            }
+            catch (Exception ex)
+            {
+                LogWriteLine($"Failed to save {_ValueName}!\r\n{ex}", LogType.Error, true);
+            }
+        }
+
+        public bool Equals(GraphicsGrade? comparedTo) => TypeExtensions.IsInstancePropertyEqual(this, comparedTo);
```

Comment:
I think since the value needs to be compared only consists an ``int`` property (``GraphicsGradeInt``), we can just call the comparison for the value instead of the entire class.
For example:
```CSharp
public bool Equals(GraphicsGrade? comparedTo) => this.GraphicsGradeInt == comparedTo?.GraphicsGradeInt;
```

Reply:
Fixed in dbac3cc41d7342f5e47a1e6c7b54c793d4d937dc